plutus-tx-plugin/test/Plugin/Basic/Spec.hs

```diff
@@ -40,5 +46,20 @@ monoK = plc (Proxy @"monoK") (\(i :: Integer) -> \(_ :: Integer) -> i)
 letFun :: CompiledCode (Integer -> Integer -> Bool)
 letFun = plc (Proxy @"letFun") (\(x::Integer) (y::Integer) -> let f z = Builtins.equalsInteger x z in f y)
 
+nonstrictLet :: CompiledCode (Integer -> Integer -> Integer)
+nonstrictLet = plc (Proxy @"strictLet") (\(x::Integer) (y::Integer) -> let z = Builtins.addInteger x y in Builtins.addInteger z z)
+
+-- GHC turns strict let-bindings into case expressions, which we correctly turn into strict let-bindings
+strictLet :: CompiledCode (Integer -> Integer -> Integer)
+strictLet = plc (Proxy @"strictLet") (\(x::Integer) (y::Integer) -> let !z = Builtins.addInteger x y in Builtins.addInteger z z)
+
+strictMultiLet :: CompiledCode (Integer -> Integer -> Integer)
+strictMultiLet = plc (Proxy @"strictLet") (\(x::Integer) (y::Integer) -> let !z = Builtins.addInteger x y; !q = Builtins.addInteger z z; in Builtins.addInteger q q)
+
+-- Here we see the wrinkles of codegen: we get non-strict external bindings for z and q, and inside that we get strict bindings
+-- corresponding to the case expressions
+strictLetRec :: CompiledCode (Integer -> Integer -> Integer)
+strictLetRec = plc (Proxy @"strictLetRec") (\(x::Integer) (y::Integer) -> let !z = Builtins.addInteger x q; !q = Builtins.addInteger y z in Builtins.addInteger z z)
```

Comment:
So it does not work in the ened in the case of strict let rec?

Reply:
And where u say 'codegen', u refer to our codegen or the ghc-core codegen?

Reply:
It sort-of works. I think the resulting output is fine, and not that far from what you would expect if you're familiar with what GHC does with bang patterns.

(I meant GHC codegen, I'll clarify)